matrix/matrix_operation.py

```diff
@@ -4,20 +4,31 @@
 
 from __future__ import annotations
 
+from collections.abc import Iterable
+from typing import Optional
+
 
 def add(*matrix_s: list[list]) -> list[list]:
```

Comment:
This is an incorrect type hint for a `*` (stared) argument. It should be `tuple[<arg1>, <arg2>, ...]` In this case, it should probably be `tuple[list[list[int]], ...]`

Also, please provide the element types for return type.